### PR TITLE
[ECExt5] Scalar Field Arithmetic

### PR DESCRIPTION
**Pull request by itzmeanjan**
This PR adds following routines computed over scalar field for `ec_ext5` curve. These will be required for implementing ECDSA signature verification over `ec_ext5` curve.

> **Note**
> `ec_ext5` scalar prime [N](https://github.com/pornin/ecgfp5/blob/82325b965f0c4043815e513954a426ba50de903b/python/ecGFp5.py#L922)

Routine | Cost ( in terms of VM cycles )
--- | --:
`mul` | 5550
`inv` | 2,809,589

Suggested review order : #495 -> `this`

### CODE REVIEW

**Comment by itzmeanjan** [comment]
Resolved merge conflicts in https://github.com/maticnetwork/miden/pull/498/commits/df93d69a77590fc00f4725cef44ce2edc876d562

File: miden/tests/integration/stdlib/math/ext5_scalar.rs
Comment (bobbinth) [inline miden/tests/integration/stdlib/math/ext5_scalar.rs:3]: minor nit: I'd combine these into a single import line.

**Review by bobbinth — CHANGES_REQUESTED** [review]
Looks great! Thank you! I left a minor nit inline, but also a couple of general comments:

1. Similar to my comment in the other PR, I think the namespace structure here could be changed a bit. Specifically, we can put all relevant modules under `std::math::ecgfp5` namespace.
2. Cycle count for inversion look really high. We should be able to reduce to under 20K cycles by relying on non-deterministic inputs. But we can leave this optimization for the future (there are also quite a few optimizations we can apply now - e.g., there are better ways to handle large number of constants now).
3. Most of the Rust code that we currently use for testing should eventually go into the [crypto](https://github.com/0xPolygonMiden/crypto) repo. But again, we can come back to this later.

File: miden/tests/integration/stdlib/math/ext5_scalar.rs
Comment (itzmeanjan) [reply miden/tests/integration/stdlib/math/ext5_scalar.rs]: Missed it, fixed in a7660d706e3c70f12196aceb5fb1fab83efd4475

**Comment by itzmeanjan** [comment]
> 1. Similar to my comment in the other PR, I think the namespace structure here could be changed a bit. Specifically, we can put all relevant modules under `std::math::ecgfp5` namespace.

To keep module structure conformant with `secp256k1` made necessary changes in 3ad5f64cacb5f1e655a2ddd9abe23ad44d4c918d

**Comment by itzmeanjan** [comment]
> 2\. Cycle count for inversion look really high. We should be able to reduce to under 20K cycles by relying on non-deterministic inputs. But we can leave this optimization for the future (there are also quite a few optimizations we can apply now - e.g., there are better ways to handle large number of constants now).

Just so that we don't forget about it https://github.com/maticnetwork/miden/issues/575

**Comment by itzmeanjan** [comment]
> 3\. Most of the Rust code that we currently use for testing should eventually go into the [crypto](https://github.com/0xPolygonMiden/crypto) repo. But again, we can come back to this later.

Yes exactly, we'll move them there, which is noted https://github.com/0xPolygonMiden/crypto/issues/24

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! My only comment is that I'd prefer to use `ecgfp5` as the namespace (instead of `ec_ext5`) as this is the official name of the curve. But other than that, we are good to go!

**Comment by itzmeanjan** [comment]
> Looks good! Thank you! My only comment is that I'd prefer to use `ecgfp5` as the namespace (instead of `ec_ext5`) as this is the official name of the curve. But other than that, we are good to go!

Agreed, changed it in https://github.com/maticnetwork/miden/pull/498/commits/9b80df8f1e4f97c20b8957288b35b30f207e99a0

**Comment by bobbinth** [comment]
Let's wait until #579 is merged. Rebasing from `next` after that should resolve all clippy errors.

**Comment by itzmeanjan** [comment]
> Let's wait until #579 is merged. Rebasing from `next` after that should resolve all clippy errors.

Rebased in https://github.com/maticnetwork/miden/pull/498/commits/1e5f7bf3a2cf3c5e2b4ad7d35a68fe02e23c1d8d

**Pull request merged**
